### PR TITLE
discovery: allow DNS seeds to be overwritten

### PR DESCRIPTION
**Pull request by guggero**
Fixes https://github.com/lightningnetwork/lnd/issues/3817.

Need to manually test on testnet, but should be pretty straightforward.

### CODE REVIEW

File: lncfg/chain.go [hunk 1]
```diff
@@ -21,4 +21,5 @@ type Chain struct {
 	BaseFee             lnwire.MilliSatoshi `long:"basefee" description:"The base fee in millisatoshi we will charge for forwarding payments on our channels"`
 	FeeRate             lnwire.MilliSatoshi `long:"feerate" description:"The fee rate used when forwarding payments on our channels. The total fee charged is basefee + (amount * feerate / 1000000), where amount is the forwarded amount."`
 	TimeLockDelta       uint32              `long:"timelockdelta" description:"The CLTV delta we will subtract from a forwarded HTLC's timelock value"`
+	DNSSeeds            []string            `long:"dnsseed" description:"The seed DNS server(s) to use for initial peer discovery. Can be specified multiple times, will overwrite the default seed servers."`
```
Comment (Roasbeef) [inline lncfg/chain.go:24]: How do we want to handle the fall back SOA address? FWIW, it's a hack we use to allow people to do the SRV requests over Tor using direct TCP DNS resolution, and isn't actually in the spec.

**Review by Roasbeef — COMMENTED** [review]
Looks pretty good to me, one comment re the fall back addresses.

File: lncfg/chain.go (hunk 1 above)
Comment (guggero) [reply lncfg/chain.go]: It'll just skip the SOA address if none was specified:
```
			if dnsSeedTuple[1] == "" {
				log.Tracef("DNS seed %v has no secondary, "+
					"skipping fallback", primarySeed)
				continue
			}
```

So non-Tor users probably don't need to specify one. I added the possibility to specify a primary and SOA fallback address as a comma separated pair.

**Review by Roasbeef — APPROVED** [review]
LGTM 🍰

**Pull request merged**